__tests__/app.test.js

```diff
@@ -156,6 +156,74 @@ describe( "/api/articles/:article_id/comments", ()=> {
           });
     })
 
+    it("POST 201 when inserts a new comment for an available selected article", () => {
+        const newComment =  {
+            body: "Testing Comments",
+            author: "lurker"
+          }
+        return request(app)
+        .post('/api/articles/9/comments')
+        .send(newComment)
+        .expect(201)
+        .then((response) => {
+            expect(response.body.comment.body).toBe("Testing Comments")
+            expect(response.body.comment.author).toBe("lurker")
+            expect(response.body.comment.comment_id).toBe(19)
+        })
+})
 
+    it("POST 400 when inserting comment to a non-exist article which violate the forienge key reference rules", () => {
+    const newComment =  {
+    body: "Testing Comments",
+    author: "lurker"
+  }
+    return request(app)
+      .post("/api/articles/10001/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+})
```

Comment:
Almost perfect - remember that a bad request is usually when there's an incorrect data type - the request is fine, it's that the article isn't found meaning that this should be a 404 instead.